### PR TITLE
improve description when expected keyfile not found

### PR DESCRIPTION
**Pull request by jasonwee**
improve description when expected keyfile not found

### CODE REVIEW

File: src/main/java/io/goobox/sync/storj/App.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
```
Comment (kaloyan-raev) [inline src/main/java/io/goobox/sync/storj/App.java]: This is unused - remove it.

File: src/main/java/io/goobox/sync/storj/App.java [hunk 2]
```diff
@@ -41,6 +42,8 @@
 import io.storj.libstorj.KeysNotFoundException;
 import io.storj.libstorj.Storj;
 
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;;
```
Comment (kaloyan-raev) [inline src/main/java/io/goobox/sync/storj/App.java]: Unused too - remove it.

File: src/main/java/io/goobox/sync/storj/App.java [hunk 3]
```diff
@@ -151,6 +154,11 @@ private void init(boolean resetAuthFile) {
         storj.setConfigDirectory(Utils.getDataDir().toFile());
         storj.setDownloadDirectory(syncDir.toFile());
 
+        if (!storj.keysExist()) {
+            logger.error("keyfile not found at the expected dir " + Utils.getDataDir().toFile());
+            System.exit(1);
```
Comment (kaloyan-raev) [inline src/main/java/io/goobox/sync/storj/App.java]: If no keys are present (the usual case after installation), the sync app will wait for a `login` command from the GUI. Exiting here will break this workflow.

File: src/main/java/io/goobox/sync/storj/App.java [hunk 4]
```diff
@@ -151,6 +154,11 @@ private void init(boolean resetAuthFile) {
         storj.setConfigDirectory(Utils.getDataDir().toFile());
         storj.setDownloadDirectory(syncDir.toFile());
 
+        if (!storj.keysExist()) {
+            logger.error("keyfile not found at the expected dir " + Utils.getDataDir().toFile());
```
Comment (kaloyan-raev) [inline src/main/java/io/goobox/sync/storj/App.java]: It would be better to improve the existing log message at line 276 than adding a new one here.

**Review by kaloyan-raev — CHANGES_REQUESTED** [review]
Nice! I think we are almost there. I left a few comments. Please, also update the PR title and description.

File: src/main/java/io/goobox/sync/storj/App.java (hunk 3 above)
Comment (jasonwee) [reply src/main/java/io/goobox/sync/storj/App.java]: ohh okay. the gui is another code base? cause when i launch this app in eclipse, i don't see any ui?

Comment (kaloyan-raev) [reply src/main/java/io/goobox/sync/storj/App.java]: https://github.com/GooBox/goobox-community-gui

File: src/main/java/io/goobox/sync/storj/App.java [hunk 5]
```diff
@@ -265,7 +265,7 @@ public void onError(int code, String message) {
                     }
                 });
             } catch (KeysNotFoundException e) {
-                logger.info("No keys found. Waiting for keys to be imported.");
+                logger.error("keyfile not found at the expected dir " + Utils.getDataDir().toFile());
```
Comment (kaloyan-raev) [inline src/main/java/io/goobox/sync/storj/App.java]: Could you not change the message completely, but just add the location?
i.e. `logger.info("No keys found at {}. Waiting for keys to be imported.", location);`

**Review by kaloyan-raev — CHANGES_REQUESTED** [review]

**Review by kaloyan-raev — APPROVED** [review]
Perfect

**Pull request merged**